src/values/AbstractValue.js

```diff
@@ -490,6 +464,14 @@ export default class AbstractValue extends Value {
       rightValues = new ValuesDomain(right);
     }
 
+    if (left.getType() === BooleanValue && right.getType() === BooleanValue) {
```

Comment:
This section is the same logic as the block starting from line325. Maybe fold into a helper function to avoid duplication?

Reply:
Also, any reason, the code here only performs this one kind of refinement, while the code you added in refineWithPathCondition() does several more?

Reply:
Good point. I don't think there should be a helper function for this. There should be just one way of constructing a logical abstract value, and it should always do the optimization.

Reply:
Agreed.